VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportPagedDataSource.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using VirtoCommerce.Domain.Catalog.Model;
+using VirtoCommerce.Domain.Catalog.Model.Search;
+using VirtoCommerce.Domain.Catalog.Services;
+using VirtoCommerce.ExportModule.Core.Model;
+using VirtoCommerce.ExportModule.Data.Services;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CatalogModule.Data.ExportImport
+{
+    public class ProductExportPagedDataSource : ExportPagedDataSource<ProductExportDataQuery, ProductSearchCriteria>
+    {
+        private readonly IBlobStorageProvider _blobStorageProvider;
+        private readonly IItemService _itemService;
+        private readonly ICatalogSearchService _catalogSearchService;
+        private readonly IBlobUrlResolver _blobUrlResolver;
+
+
+        public ProductExportPagedDataSource(IBlobStorageProvider blobStorageProvider, IItemService itemService, ICatalogSearchService catalogSearchService, IBlobUrlResolver blobUrlResolver, ProductExportDataQuery dataQuery) : base(dataQuery)
+        {
+            _blobStorageProvider = blobStorageProvider;
+            _itemService = itemService;
+            _catalogSearchService = catalogSearchService;
+            _blobUrlResolver = blobUrlResolver;
+        }
+
+
+        protected override ExportableSearchResult FetchData(ProductSearchCriteria searchCriteria)
+        {
+            var result = Array.Empty<CatalogProduct>();
+            int totalCount = 0;
+
+            var responseGroup = BuildResponseGroup();
+            var productIds = searchCriteria.ObjectIds?.ToArray() ?? Array.Empty<string>();
+
+            if (productIds.IsNullOrEmpty())
+            {
+                var catalogSearchCriteria = AbstractTypeFactory<SearchCriteria>.TryCreateInstance();
+                catalogSearchCriteria.SearchInChildren = true;
```

Comment:
Shouldn't it go from `searchCriteria`?

Reply:
Criteria and Catalog filters work onliy when SearchInChildren is true. So when we open export blade from catalog module and SearchInChildren is false as default results are always empty. For better UI experience we should set SearchInChildren to true.